rwslib/rws_requests/__init__.py

```diff
@@ -65,14 +76,21 @@ def __eq__(self, other):
     def __ne__(self, other):
         return not self.__eq__(other)
 
-    def result(self, request):
-        """Process a result to create a custom output"""
+    def result(self, response):
+        """
+        Process a result to create a custom output
+        :param requests.models.Response response: returned response
+        :return:
+        """
         # By default return text
-        return request.text
+        response.encoding = "utf-8-sig"
```

Comment:
Ah, to my earlier comment. You're ahead of me. Request is already responsible for this.

Reply:
Yeah, I also changed the variable to response as that more accurately represents the object in play.

Reply:
So my point is that each request type should know what kind of encoding it has. It may be an XML response type with a utf-8 BOM in which case (as here) it sets it's response.encoding for its own return value. But in the rws connection at the moment, once a request has been fired off there's:

```
        self.last_result = r #see also r.elapsed for timedelta object.
        self.last_result = r  # see also r.elapsed for timedelta object.
    
     # ensure the response.text is usable
        r.encoding = "utf-8-sig"
```
And that last line seem to me dangerous because we don't know that the response.encoding is definitely utf-8-sig - what if a custom dataset response returns UTF-32 content? So I think rwsconnection has no place setting this and it's solely the responsibility of the request type it's firing off to know what it should return.

I hope I am making sense...

Reply:
Of course you are making sense.

Ok, that shouldn't have been there.  Have removed, I had that in too many places.  "utf-8-sig" transparently removes the BOM from the response. 

The response object has the `.text` which is supposed to return the content as a unicode which is what you have used.   However, we see that it didn't detect the encoding correctly for a request including Japanese characters; in this case the content-type was 'text/xml' (FormDatasetRequest), which Postman seems to handle, but requests did not.

This may be down to how Rave declares (or otherwise) the content-type for the core RWS and CDS responses; I've see some with text/xml, text/csv; encoding=utf-8, '' (yeah, nothing).  

Just to be clear, you want me to remove any attempt to work around the encoding and let requests handle it?  If they don't get what they expect they can access the last_result and access and decode the response.text manually?  Once you give me the nod, I'll back all the changes out.  
Thanks!

Reply:
OK, thanks for explanation. Background is clear. When I say "request" I mean the RWSRequest base class. At the moment (in this MR) the result of a call is returned like this:

```
    def result(self, response):
        """
        Process a result to create a custom output
        :param requests.models.Response response: returned response
        :return:
        """
        # By default return text
        response.encoding = "utf-8-sig"
        return response.text
```
That is fine but assumes the default of any non-overridden RWSRequest class result is utf-8-sig encoded. I'm saying that any RWSRequest class is responsible for declaring / setting the correct response.encoding it its own result() method, overriding the base one (which assumes utf-8-sig encoding) if it needs to. i.e. if I have a custom dataset that returns UTF-32 encoded characters then it's up to me to override the request result() method to set that.

I think that means  that besides removing the forced setting of utf-8-sig encoding in the rwsconnection class there's nothing else for you to do.